XAudio2/Common/XAudio2Versions.h

```diff
@@ -0,0 +1,42 @@
+//--------------------------------------------------------------------------------------
+// File: XAudio2Versions.h
+//
+// Defines preprocessor symbols to distinguish between different versions of XAudio2,
+// and includes the appropriate header files for the XAudio2 version used.
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//--------------------------------------------------------------------------------------
+#pragma once
+
+// In case _WIN32_WINNT is not already defined, this header will define it to the highest
+// version supported by the Windows SDK.
+#include <sdkddkver.h>
+
+// Set preprocessor symbols for the different versions of XAudio2.
+#ifdef USING_XAUDIO2_REDIST
```

Comment:
If I understand correctly, ``USING_XAUDIO2_REDIST`` is not set in the projects or the source, but by the NuGet package itself being in the project. Nice.